### DISCUSSION TITLE
Test capabilities for FastAPI apps ran with ui.run_with

### DISCUSSION THREAD

**Discussion by mbrulatout** [zauberzeug/nicegui#2313] [Ideas / Feature Requests]
Hello,

For `ui.run` apps, the screen class provides the appropriate features to test your current app (see [here](https://github.com/zauberzeug/nicegui/blob/main/tests/test_api_router.py) ).

I'm running NiceGUI through `ui.run_with` and as it stands I'm unable to run any UI test.

There seems to be a bit of rework in [`screen.py`](https://github.com/zauberzeug/nicegui/blob/main/nicegui/testing/screen.py) class at the very least as it calls `ui.run`

**Comment by falkoschindler** [comment]
Yes, @mbrulatout, so far we used the `Screen` class only for testing apps based on `ui.run`, not `ui.run_with`. Adding support for the latter would be great. But we could certainly need some help. Anyone interested in giving it a go? 🙂
